### ISSUE TITLE
By default telegraf.conf for Windows binaries contain Linux input plugin configurations

### ISSUE THREAD

**Issue by artavazdg** [influxdata/telegraf#10162]
### Relevent telegraf.conf

```toml
# Read metrics about cpu usage
[[inputs.cpu]]
  ## Whether to report per-cpu stats or not
  percpu = true
  ## Whether to report total system cpu stats or not
  totalcpu = true
  ## If true, collect raw CPU time metrics
  collect_cpu_time = false
  ## If true, compute and report the sum of all non-idle CPU states
  report_active = false


# Read metrics about disk usage by mount point
[[inputs.disk]]
  ## By default stats will be gathered for all mount points.
  ## Set mount_points will restrict the stats to only the specified mount points.
  # mount_points = ["/"]

  ## Ignore mount points by filesystem type.
  ignore_fs = ["tmpfs", "devtmpfs", "devfs", "iso9660", "overlay", "aufs", "squashfs"]


# Read metrics about disk IO by device
[[inputs.diskio]]
  ## By default, telegraf will gather stats for all devices including
  ## disk partitions.
  ## Setting devices will restrict the stats to the specified devices.
  # devices = ["sda", "sdb", "vd*"]
  ## Uncomment the following line if you need disk serial numbers.
  # skip_serial_number = false
  #
  ## On systems which support it, device metadata can be added in the form of
  ## tags.
  ## Currently only Linux is supported via udev properties. You can view
  ## available properties for a device by running:
  ## 'udevadm info -q property -n /dev/sda'
  ## Note: Most, but not all, udev properties can be accessed this way. Properties
  ## that are currently inaccessible include DEVTYPE, DEVNAME, and DEVPATH.
  # device_tags = ["ID_FS_TYPE", "ID_FS_USAGE"]
  #
  ## Using the same metadata source as device_tags, you can also customize the
  ## name of the device via templates.
  ## The 'name_templates' parameter is a list of templates to try and apply to
  ## the device. The template may contain variables in the form of '$PROPERTY' or
  ## '${PROPERTY}'. The first template which does not contain any variables not
  ## present for the device is used as the device name tag.
  ## The typical use case is for LVM volumes, to get the VG/LV name instead of
  ## the near-meaningless DM-0 name.
  # name_templates = ["$ID_FS_LABEL","$DM_VG_NAME/$DM_LV_NAME"]


# Get kernel statistics from /proc/stat
[[inputs.kernel]]
  # no configuration


# Read metrics about memory usage
[[inputs.mem]]
  # no configuration


# Get the number of processes and group them by status
[[inputs.processes]]
  # no configuration


# Read metrics about swap memory usage
[[inputs.swap]]
  # no configuration


# Read metrics about system load & uptime
[[inputs.system]]
  ## Uncomment to remove deprecated metrics.
  # fielddrop = ["uptime_format"]
```

### System info

1.20.4, Windows Server

### Steps to reproduce

1. download telegraf binaries from "https://dl.influxdata.com/telegraf/releases/telegraf-1.20.4_windows_amd64.zip" URL, which is mentioned on telegraf official documentation page, to Windows endpoint
2. extract files in the necessary location
3. open telegraf.conf file

### Expected behavior

By default "telegraf.conf" for Windows binaries should contain Windows input plugin configurations.

### Actual behavior

By default "telegraf.conf" for Windows binaries contain Linux input plugin configurations. they are mentioned in "Relevent telegraf.conf".

### Additional info

Till 1.20.x version, by default "telegraf.conf" for Windows binaries contained Windows input plugin configurations.

**Comment by powersj** [comment]
Hi,

Of the above configuration, only the process input does not appear to be supported:

```s
2022-01-20T22:14:31Z W! [inputs.processes] Current platform is not supported
diskio,host=DESKTOP-6BC121I,name=C: reads=92559i,writes=40938i,read_bytes=2428771328i,write_time=9i,io_time=0i,weighted_io_time=0i,iops_in_progress=0i,merged_reads=0i,write_bytes=876824576i,read_time=57i,merged_writes=0i 1642716880000000000
mem,host=DESKTOP-6BC121I used_percent=54.25058142057845,available_percent=45.74941857942155,total=8589389824i,available=3929595904i,used=4659793920i 1642716880000000000
system,host=DESKTOP-6BC121I load15=0,n_cpus=8i,load1=0,load5=0 1642716880000000000
system,host=DESKTOP-6BC121I uptime=121i 1642716880000000000
system,host=DESKTOP-6BC121I uptime_format=" 0:02" 1642716880000000000
disk,device=C:,fstype=NTFS,host=DESKTOP-6BC121I,mode=rw,path=\C: free=68207128576i,used=34282840064i,used_percent=33.449946876674154,inodes_total=0i,inodes_free=0i,inodes_used=0i,total=102489968640i 1642716880000000000
swap,host=DESKTOP-6BC121I used=4618833920i,free=5983821824i,used_percent=43.5629905518132,total=10602655744i 1642716880000000000
swap,host=DESKTOP-6BC121I in=0i,out=0i 1642716880000000000
```

The configurations between Windows and Linux are already different. And by different, I saw things like `bcache` only in the linux one. This is determined by build flags.

The process plugin is allowed currently because the files for process print out a helpful error about this:

```go
e.Log.Warn("Current platform is not supported")
```

Are there any other inputs that you came across with this issue?

**Comment by srebhan** [comment]
I think in general this is a valid point (also for other platforms). I'm reopening this as this is not yet fixed but I plan to fix it once #11654 is merged.

**Comment by powersj** [comment]
After further discussions, we decided that the default config would have all plugins listed, regardless of the platform. We did go add in comments by each plugin that is specific to a platform to highlight that it might not be available.

Closing.
